coderd/coderd.go

```diff
@@ -444,6 +448,38 @@ func New(options *Options) *API {
 	if err != nil {
 		panic(xerrors.Errorf("get deployment ID: %w", err))
 	}
+
+	fetcher := &cryptokeys.DBFetcher{
+		DB: options.Database,
+	}
+
+	if options.OIDCConvertKeyCache == nil {
+		options.OIDCConvertKeyCache, err = cryptokeys.NewSigningCache(ctx,
+			options.Logger.Named("oidc_convert_keycache"),
+			fetcher,
+			codersdk.CryptoKeyFeatureOIDCConvert,
+		)
+		must(options.Logger, "start oidc convert key cache", err)
```

Comment:
Why not? I can't think of a good reason why trying to query keys should fail which means there's probably something wrong with our DB connection.

Reply:
In distributed systems you want to minimize coupling and self-heal.  A particularly annoying sort of coupling is the order in which subcomponents start up.  That is, you want it to be the case that things work correctly if coderd and the database both are up, regardless of whether coderd started before the database. Ordering dependencies can be a nightmare as you add more systems. It might not just be a database server: you might also need the DNS route to have been propagated, or an authenticating proxy to be up, etc.

Reply:
I realize there are other things that will fail if the DB is unreachable, but let's not make our problems worse.